### PR TITLE
[AIRFLOW-5885] List of tests is generated dynamically when you enter Breeze

### PR DESCRIPTION
**Pull request by potiuk**
The list of tests does not change that often, yet it could not be generated
in Docker before all files were added (due to imports etc.). Therefore it is much better
solution to generate the list of tests in pre-commit - only when any of the test .py files
change and commit it together with the sources.

Also - since we are modifying Dockerfile, it turned out that "airflow" user is not needed
in the CI image any more so I removed it to save additional Dockerfile modification.

Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5885
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Comment by potiuk** [comment]
One of the next steps towards simpler / better image + production one. This make the CI images build faster + we have list of all tests automatically updated via pre-commit every time a new test is added rather than during every docker build.

**Comment by mik-laj** [comment]
Is it not possible to generate this file at another time, e.g. when starting the breeze environment?  I am afraid that the existence of this file will cause many problems, because it will need to be updated forever. I would prefer that every change changes only what is necessary.  This makes it very difficult for beginners to add new functions because they may not have pre-commit installed. pre-commit is optional, but it is recommended to use it. The breeze environment is also optional, but it is recommended to use it. Changes to patches should also be possible for people who do not want to use the Breeze environment. Airflow is used by many companies and they may have their own forks, so additional files in the patch will hinder cherry-pick.

BTW. Personally, I don't use automatic completion of test names, because I prefer to specify them by file name, and this is provided by Bash, so the necessity of completing a file that I don't use would cause a lot of emotion if it caused problems.

**Comment by potiuk** [comment]
Side comment: I think eventually everyone should use pre-commit. It makes it so much easier to keep the source code in order and to fix many of the problems reported by Travis even before they are reported. We should strongly encourage people to use it. I think we should encourage good practices rather than discourage them.

But I agree having the .txt file in the repo is kind of superfluous. I think it is a good idea to generate it when you enter Breeze. I can run it in background.

**Comment by potiuk** [comment]
@mik-laj  -> test list is now dynamically generated when you enter Breeze. It was a good idea to do it this way :).

File: scripts/ci/in_container/run_extract_tests.sh [hunk 1]
```diff
@@ -16,24 +16,27 @@
...
-gosu "${AIRFLOW_USER}" \
-    python "${AIRFLOW_SOURCES}/tests/test_utils/get_all_tests.py" \
-                    "${HOME}/all_tests.xml" >"${HOME}/all_tests.txt"; \
+nosetests --collect-only --with-xunit --xunit-file="${TMP_FILE}"
 
-echo ". ${HOME}/.bash_completion" >> "${HOME}/.bashrc"
+export AIRFLOW_ALL_TESTS="${HOME}/all_tests.txt"
 
-chmod +x "${HOME}/run-tests-complete"
+<"${AIRFLOW_SOURCES}/license-templates/LICENSE.txt" sed 's/^/# /' | sed 's/# $/#/'  > "${AIRFLOW_ALL_TESTS}"
```
Comment (mik-laj) [inline scripts/ci/in_container/run_extract_tests.sh]: Do we need a license in this file?

File: scripts/ci/local_ci_fix_ownership.sh [hunk 2]
```diff
@@ -30,7 +30,7 @@ basic_sanity_checks
 
 script_start
 
-export PYTHON_VERSION=${PYTHON_VERSION:="3.6"}
+export PYTHON_VERSION=3.5
```
Comment (mik-laj) [inline scripts/ci/local_ci_fix_ownership.sh:33]: Does this mean that I will have two images?

File: scripts/ci/in_container/run_extract_tests.sh (hunk 1 above)
Comment (potiuk) [reply scripts/ci/in_container/run_extract_tests.sh]: Right! We don't

File: scripts/ci/local_ci_fix_ownership.sh (hunk 2 above)
Comment (potiuk) [reply scripts/ci/local_ci_fix_ownership.sh:33]: By default we have python 3.5 used for all static tests - I synchronised it to use 3.5 everywhere (and we do not rely on local python version any more). So you will have just one image (3.5) unless you explicitly choose --python-version= in breeze or set PYTHON_VERSION variable.

**Review by mik-laj — APPROVED** [review]

**Pull request merged**